### PR TITLE
add presubmit CI job for ec2+GPU scenario

### PR DESCRIPTION
**Pull request by dims**
Add a presubmit so it's easier to test the ec2/AWS GPU scenarios.

### CODE REVIEW

File: config/jobs/kubernetes/sig-cloud-provider/aws/ec2-e2e.yaml [hunk 1]
```diff
@@ -242,6 +242,75 @@ presubmits:
             requests:
               cpu: 8
               memory: 10Gi
+  - name: pull-kubernetes-e2e-ec2-device-plugin-gpu
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-cloud-provider/aws/ec2-e2e.yaml:245]: nit: can we please leave `pull-kubernetes-*` for k/k? usually it's `pull-$repo`

I thought for a moment we were adding a new always_run k/k presubmit

Comment (dims) [reply config/jobs/kubernetes/sig-cloud-provider/aws/ec2-e2e.yaml:245]: Sorry my bad. fixing.

**Review by BenTheElder — COMMENTED** [review]
/lgtm

**Pull request merged**